ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java

```diff
@@ -378,6 +378,44 @@ protected void updateResolvedContainers(Account account, Collection<Container> r
     }
   }
 
+  @Override
+  public Container getContainer(String accountName, String containerName) throws AccountServiceException {
+    Account account = getAccountByName(accountName);
+    if (account == null) {
+      return null;
+    }
+    Container container = account.getContainerByName(containerName);
+    if (container == null) {
+      // If container is not present in the account, refresh cache and retry
+      try {
+        fetchAndUpdateCache();
+        container = getAccountByName(accountName).getContainerByName(containerName);
+      } catch (SQLException e) {
+        throw translateSQLException(e);
+      }
+    }
+    return container;
+  }
+
+  @Override
+  public Container getContainer(String accountName, Short containerId) throws AccountServiceException {
+    Account account = getAccountByName(accountName);
+    if (account == null) {
+      return null;
+    }
+    Container container = account.getContainerById(containerId);
+    if (container == null) {
+      // If container is not present in the account, refresh cache and retry
+      try {
+        fetchAndUpdateCache();
```

Comment:
I was thinking we could query the individual container record.  A bit lighter weight, and also a guard against any issues that might crop up with the sync-latest query (isolation, corrupted timestamp).